lms/djangoapps/discussion_api/api.py

```diff
@@ -81,6 +82,8 @@ def _get_course(course_key, user):
         course = get_course_with_access(user, 'load', course_key, check_if_enrolled=True)
     except Http404:
         raise CourseNotFoundError("Course not found.")
+    except CourseAccessRedirect:
+        raise CourseNotFoundError("Course not found.")
```

Comment:
I've added a comment. I don't want to introduce new logging because I don't know how often this endpoint will raise those exceptions.